### PR TITLE
Merge `tox.ini` and `setup.py` into `pyproject.toml`

### PR DESCRIPTION
**Pull request by ruestefa**
- Move configs of `pytest`, `isort`, `flake8` (by aid of `flake8-pyproject`), `pylint` and `pydocstyle` from `tox.ini` to `pyproject.toml`
- Remove `tox.ini` and comment `tox*` in `dev-requirements.yml`
  - `tox` not set up to test different python versions (separate task #49), so it's not very useful
- Note: Already contains changes in #50
- Update: Merged #47

### CODE REVIEW

**Comment by ruestefa** [comment]
Add `run-mypy.sh` script to run `mypy` and use it in `pre-commit` and github actions. Motivation: `mypy` should always be applied to the whole code base, not just over the changed files, changes can introduce bugs in other files. (In addition, running `mypy` locally (i.e., in the active dev env, not in an isolated pre-commit env) makes all deps available to mypy, which otherwise would all have to be installed in the pre-commit env.)

**Review by regDaniel — CHANGES_REQUESTED** [review]
Again, quick discussion on code overhead, in terms of technical content I agree.

File: {{cookiecutter.project_slug}}/test_all.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (ruestefa) [reply {{cookiecutter.project_slug}}/test_all.sh:1]: Removed.

File: {{cookiecutter.project_slug}}/requirements/dev-requirements.yml [hunk 2]
```diff
@@ -20,5 +19,7 @@ dependencies :
   - rstcheck
   - sphinx
   - sphinx-autobuild
-  - tox
-  - tox-conda
\ No newline at end of file
+  # - tox
```
[the remark this answers is not in the document]
Comment (ruestefa) [reply {{cookiecutter.project_slug}}/requirements/dev-requirements.yml:22]: Deleted.

**Comment by ruestefa** [comment]
Merged #47 here and closed that PR.

**Review by regDaniel — APPROVED** [review]

**Comment by regDaniel** [comment]
I think we can merge what is in this PR.

**Comment by cosunae** [comment]
launch jenkins

**Comment by cosunae** [comment]
launch jenkins

**Comment by cosunae** [comment]
I fixed jenkins for the blueprint. The one of the project (template) is still broken, not adapted to latest changes of the install. But since it is not tested here, I will create an issue to fix that later. 
I will merge

**Pull request merged**